src/route/index.ts

```diff
@@ -451,6 +452,53 @@ export async function initApiServer(
         },
       });
 
+      instance.route({
+        method: "GET",
+        url: "/is-sac-contract/:contractId",
+        schema: {
+          params: {
+            ["contractId"]: {
+              type: "string",
+              validator: (qStr: string) => isContractId(qStr),
+            },
+          },
+          querystring: {
+            ["network"]: {
+              type: "string",
+              validator: (qStr: string) => isNetwork(qStr),
+            },
+          },
+        },
+        handler: async (
+          request: FastifyRequest<{
+            Params: { ["contractId"]: string };
+            Querystring: {
+              ["network"]: NetworkNames;
+            };
+          }>,
+          reply
+        ) => {
+          const contractId = request.params["contractId"];
+          const { network } = request.query;
+
+          const skipSorobanPubnet = network === "PUBLIC" && !useSorobanPublic;
+          if (skipSorobanPubnet) {
+            return reply.code(400).send("Soroban has been disabled on pubnet");
+          }
+
+          try {
+            const isSacContract = await isSacContractExecutable(
+              contractId,
+              network
+            );
+
+            reply.code(200).send({ isSacContract });
+          } catch (error) {
+            reply.code(500).send("Unexpected Server Error");
```

Comment:
since `isSacContractExecutable` can throw for legitimate errors, I don't think we want a 500 in this catch block. For example if you pass it a contract ID for an expired contract and it can't fetch the entries, it will throw an `ERROR.ENTRY_NOT_FOUND.CONTRACT_CODE` error.

Reply:
Maybe it makes more sense not to throw in that helper?